### PR TITLE
Create `pvlib.iotools.get_solrad`

### PR DESCRIPTION
**Pull request by kandersolar**
- ~[ ] Closes #xxxx~
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

[SOLRAD](https://gml.noaa.gov/grad/solrad/index.html) is a network of ground stations in the US similar to SURFRAD.  SOLRAD data is available on the NOAA website in the form of daily files which pvlib-python can fetch and parse using `read_solrad`.  However, this requires the user to construct each daily file URL manually and iteratively retrieve files, which is a minor annoyance.

This PR creates a `get_solrad` function to wrap the existing `read_solrad` function.  It constructs the necessary filenames based on the supplied date range, iteratively calls `read_solrad`, returns a single concatenated dataframe.

### CODE REVIEW

File: pvlib/iotools/solrad.py [hunk 1]
```diff
@@ -121,3 +123,75 @@ def read_solrad(filename):
...
+    end = pd.to_datetime(end)
+
+    # Generate list of filenames
+    dates = pd.date_range(start, end, freq='d', inclusive='both')
+    years = dates.year
+    daysofyear = dates.dayofyear
+    filenames = [
+        f"{station}/{year}/{station}{year-2000}{doy:03}.dat"
+        for year, doy in zip(years, daysofyear)
+    ]
```
Comment (AdamRJensen) [inline pvlib/iotools/solrad.py:182]:
```suggestion
    dates = pd.date_range(start, end, freq='d', inclusive='both')
    filenames = [
        f"{station}/{d.year}/{station}{d.strftime('%y')}{d.dayofyear:03}.dat"
        for d in dates
    ]
```
Personally I would find this easier.

File: pvlib/iotools/solrad.py [hunk 2]
```diff
@@ -121,3 +123,75 @@ def read_solrad(filename):
...
+    Notes
+    -----
+    Recent SOLRAD data is 1-minute averages.  Prior to 2015-01-01, it was
+    3-minute averages.
+
+    Examples
+    --------
+    >>> # Retrieve two months irradiance data from the ABQ SOLRAD station
+    >>> data, metadata = pvlib.iotools.get_solrad(
+    >>>     start="2020-01-01", end="2020-01-31", station='abq')
```
Comment (AdamRJensen) [inline pvlib/iotools/solrad.py:164]:
```suggestion
    >>>     station='abq', start="2020-01-01", end="2020-01-31")
```
Just to match the order of the inputs

File: pvlib/iotools/solrad.py [hunk 3]
```diff
@@ -121,3 +123,75 @@ def read_solrad(filename):
...
+    # Use pd.to_datetime so that strings (e.g. '2021-01-01') are accepted
+    start = pd.to_datetime(start)
+    end = pd.to_datetime(end)
+
+    # Generate list of filenames
+    dates = pd.date_range(start, end, freq='d', inclusive='both')
+    years = dates.year
+    daysofyear = dates.dayofyear
+    filenames = [
+        f"{station}/{year}/{station}{year-2000}{doy:03}.dat"
```
Comment (AdamRJensen) [inline pvlib/iotools/solrad.py:180]: On the website, the station abbreviations are written in uppercase, so I suggest we add ``station.lower()``

**Review by AdamRJensen — COMMENTED** [review]
Would be nice with a "See Also" section in both the read and get function documentation.

File: pvlib/iotools/solrad.py [hunk 4]
```diff
@@ -121,3 +127,78 @@ def read_solrad(filename):
...
+        for d in dates
+    ]
+
+    dfs = []  # Initialize list of monthly dataframes
+    for f in filenames:
+        try:
+            dfi = read_solrad(url + f)
+            dfs.append(dfi)
+        except urllib.error.HTTPError:
+            warnings.warn(f"The following file was not found: {f}")
```
Comment (AdamRJensen) [inline pvlib/iotools/solrad.py:197]: Why is this listed as not covered? It seems like this it should be covered by ``test_get_solrad_missing_day`

Comment (kandersolar) [reply pvlib/iotools/solrad.py:197]: I agree, that line should be counted as covered.  The PR checks are showing 100% coverage for me, not sure why/how it is showing uncovered for you.

**Review by AdamRJensen — APPROVED** [review]
Nicely done!

Leaving this PR open for about a weeks time for anyone else to review

**Comment by kandersolar** [comment]
@AdamRJensen this PR is now updated to account for #1968.  It also fixes an issue with #1968 where trying to `read_solrad` a url that 404s raised `ValueError: could not convert string to float: '<html'` instead of `HTTPError: 404 Client Error`. 

Take a look and merge if it looks good to you?

**Pull request merged**